Snowflake.Data.Tests/IntegrationTests/SFBindTestIT.cs

```diff
@@ -769,5 +774,290 @@ public void testPutArrayBind1()
+
+                // Assert
+                var row = 0;
+                using (var select = conn.CreateCommand($"select id, ts from {TableName} order by id"))
+                {
+                    s_logger.Debug(select.CommandText);
+                    var reader = select.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        ++row;
+                        string faultMessage = $"Mismatch for row: {row}, {testCase}";
+                        Assert.AreEqual(row, reader.GetInt32(0));
+                        expected.IsEqual(reader.GetValue(1), comparisonFormat, faultMessage);
+                    }
+                }
+                Assert.AreEqual(1+smallBatchRowCount+bigBatchRowCount, row);
+            }
+            
+            void InsertSingleRecord(IDbConnection conn, string sqlInsert, DbType binding, int identifier, ExpectedTimestampValueProvider ts)
+            {
+                using (var insert = conn.CreateCommand(sqlInsert))
+                {
+                    // Arrange
+                    insert.Add("1", DbType.Int32, identifier);
+                    insert.Add("2", binding, ts);
+                    
+                    // Act
+                    s_logger.Info(sqlInsert);
+                    var rowsAffected = insert.ExecuteNonQuery();
+                    
+                    // Assert
+                    Assert.AreEqual(1, rowsAffected);
+                    Assert.IsNull(((SnowflakeDbCommand)insert).GetBindStage());
+                }
+            }
+
+            void InsertMultipleRecords(IDbConnection conn, string sqlInsert, DbType binding, int initialIdentifier, ExpectedTimestampValueProvider ts, int rowsCount, bool shouldUseBinding)
+            {
+                using (var insert = conn.CreateCommand(sqlInsert))
+                {
+                    // Arrange
+                    insert.Add("1", DbType.Int32, Enumerable.Range(initialIdentifier, rowsCount).ToArray());
+                    insert.Add("2", binding, ts, rowsCount);
+                    
+                    // Act
+                    s_logger.Debug(sqlInsert);
+                    var rowsAffected = insert.ExecuteNonQuery();
+                    
+                    // Assert
+                    Assert.AreEqual(rowsCount, rowsAffected);
+                    if (shouldUseBinding)
+                        Assert.IsNotEmpty(((SnowflakeDbCommand)insert).GetBindStage());
+                    else
+                        Assert.IsNull(((SnowflakeDbCommand)insert).GetBindStage());
+                }
+            }
+        }
+
+        private static string ColumnTypeWithPrecision(SFDataType columnType, Int32? columnPrecision) 
+            => columnPrecision != null ? $"{columnType}({columnPrecision})" : $"{columnType}";
+    }
+    
+    class ExpectedTimestampValueProvider
```

Comment:
Provider should provide something, but this class does not provide anything, it is rather a test case. What do you think about the name TimeoutTestCase?

Reply:
It does provide an expected value in the required by columnType format which can be either DateTime or DateTimeOffset. Doesn't it make sense?

Reply:
And it is not a TestCase alone. I could get rid of this class and in the result some of the new test lines
require an if expression related to columnType, SFDataType etc

Reply:
Changed as agreed